python/cudf/cudf/tests/test_binops.py

```diff
@@ -877,6 +877,67 @@ def test_binop_bool_uint(func, rhs):
     )
 
 
+@pytest.mark.parametrize(
+    "series_dtype", (np.bool_, np.int8, np.uint8, np.int64, np.uint64)
+)
+@pytest.mark.parametrize(
+    "divisor_dtype",
+    (
+        pytest.param(
+            np.bool_,
+            marks=pytest.mark.xfail(
+                reason=(
+                    "Pandas handling of division by zero-bool is too strange"
```

Comment:
I don't really like the CUDF handling either if both operands are series because it's inconsistent:

```
In [59]: c // cudf.Series([True, True])
Out[59]: 
0     True
1    False
dtype: bool

In [60]: c // cudf.Series([False, False])
Out[60]: 
0    True
1    True
dtype: bool

In [61]: c // np.bool_(True)
Out[61]: 
0    1
1    0
dtype: int8

In [62]: c // np.bool_(False)
Out[62]: 
0    inf
1    NaN
dtype: float64

In [63]: c
Out[63]: 
0     True
1    False
dtype: bool
```

Similarly for modulo:
```
In [54]: c
Out[54]: 
0     True
1    False
dtype: bool

In [55]: c % cudf.Series([True, True])
Out[55]: 
0    False
1    False
dtype: bool

In [56]: c % cudf.Series([False, False])
Out[56]: 
0    False
1    False
dtype: bool

In [57]: c % np.bool_(True)
Out[57]: 
0    0
1    0
dtype: int8

In [58]: c % np.bool_(False)
Out[58]: 
0   NaN
1   NaN
```

Reply:
I think we can make this work consistently with column vs. scalar with:

```patch
diff --git a/python/cudf/cudf/core/column/numerical.py b/python/cudf/cudf/core/column/numerical.py
index bf0f6776d4..44a92fef6e 100644
--- a/python/cudf/cudf/core/column/numerical.py
+++ b/python/cudf/cudf/core/column/numerical.py
@@ -223,7 +223,6 @@ class NumericalColumn(NumericalBaseColumn):
                 # Guard against division by zero for integers.
                 if (
                     (tmp.dtype.type in int_float_dtype_mapping)
-                    and (tmp.dtype.type != np.bool_)
                     and (
                         (
                             (
```

That line appeared in https://github.com/rapidsai/cudf/pull/10421/commits/ed86fdff83dd73bca443998be1c1cc0e311063bf (part of #10421) @vyasr any recollection why the special-casing to not promote to float when dividing by a boolean zero?

Reply:
In pandas, we've now restricted the floordiv case to raise a `NotImplementedError`

https://github.com/pandas-dev/pandas/blob/d13c9e034ce8a1d738766c4b1cec80c76f5523be/pandas/core/ops/array_ops.py#L513.

Maybe pandas should restrict modulo too since it involves division

Reply:
The booleans form a finite field (GF(2)) so all of these operations are well defined (excepting division by zero) so this is a somewhat odd decision.